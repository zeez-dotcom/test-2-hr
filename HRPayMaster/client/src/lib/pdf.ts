import pdfMake from 'pdfmake/build/pdfmake.js';
import pdfFonts from 'pdfmake/build/vfs_fonts.js';
import type { TDocumentDefinitions } from 'pdfmake/interfaces';

pdfMake.vfs = pdfFonts as any;

export const sanitizeString = (str: string): string =>
  str.replace(/[&<>"']/g, c => {
    switch (c) {
      case '&':
        return '&amp;';
      case '<':
        return '&lt;';
      case '>':
        return '&gt;';
      case '"':
        return '&quot;';
      case "'":
        return '&#39;';
      default:
        return c;
    }
  });

function sanitize(obj: any): any {
  if (typeof obj === 'string') return sanitizeString(obj);
  if (Array.isArray(obj)) return obj.map(sanitize);
  if (obj instanceof Date) { return obj; }
  if (obj && typeof obj === 'object') {
    const res: any = {};
    for (const [key, value] of Object.entries(obj)) {
      res[key] = sanitize(value);
    }
    return res;
  }
  return obj;
}

export interface EmployeeLite {
  firstName: string;
  lastName: string;
  id: string;
  position?: string | null;
}

export interface EmployeeEventLite {
  title: string;
  eventDate: string | Date;
  amount?: string | null;
}

export function buildEmployeeReport(data: { employee: EmployeeLite; events: EmployeeEventLite[] }): TDocumentDefinitions {
  const { employee, events } = data;
  return {
    info: { title: `${employee.firstName} ${employee.lastName} Report` },
    content: [
      { text: `${employee.firstName} ${employee.lastName}`, style: 'header' },
      employee.position ? { text: employee.position } : '',
      { text: `ID: ${employee.id}`, margin: [0, 0, 0, 10] },
      { text: 'Events', style: 'subheader' },
      {
        table: {
          headerRows: 1,
          widths: ['*', 'auto'],
          body: [
            ['Title', 'Date'],
            ...events.map(e => [e.title, new Date(e.eventDate).toISOString().split('T')[0]]),
          ],
        },
      },
    ],
    styles: {
      header: { fontSize: 18, bold: true },
      subheader: { fontSize: 14, bold: true, margin: [0, 10, 0, 5] },
    },
  };
}

export function buildEmployeeHistoryReport(employees: EmployeeLite[]): TDocumentDefinitions {
  return {
    info: { title: 'Employee History Report' },
    content: [
      { text: 'Employee History Report', style: 'header' },
      {
        table: {
          headerRows: 1,
          widths: ['*', 'auto'],
          body: [
            ['Name', 'ID'],
            ...employees.map(e => [`${e.firstName} ${e.lastName}`, e.id]),
          ],
        },
      },
    ],
    styles: {
      header: { fontSize: 18, bold: true },
    },
  };
}

export function openPdf(docDefinition: TDocumentDefinitions) {
  const sanitized = sanitize(docDefinition) as TDocumentDefinitions;
  pdfMake.createPdf(sanitized).open();
}

export function pdfBuffer(docDefinition: TDocumentDefinitions): Promise<Uint8Array> {
  const sanitized = sanitize(docDefinition) as TDocumentDefinitions;
  return new Promise(resolve => {
    pdfMake.createPdf(sanitized).getBuffer((buffer: Uint8Array) => {
      resolve(buffer);
    });
  });
}
