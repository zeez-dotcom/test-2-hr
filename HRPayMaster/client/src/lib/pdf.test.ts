import { describe, it, expect } from 'vitest';
import { pdfBuffer, sanitizeString, buildEmployeeReport, buildEmployeeHistoryReport } from './pdf';

describe('pdf utility', () => {
  it('sanitizes strings', () => {
    expect(sanitizeString('<script>alert(1)</script>')).toBe('&lt;script&gt;alert(1)&lt;/script&gt;');
  });

  it('generates pdf buffer', async () => {
    const buffer = await pdfBuffer({
      content: [{ text: '<b>hello</b>' }],
      info: { title: 'Test', creationDate: new Date(0) }
    });
    expect(Buffer.from(buffer).toString('base64')).toMatchSnapshot();
  });

  it('sanitizes nested structures', async () => {
    const buffer = await pdfBuffer({
      content: [
        { text: '<img src=x onerror=alert(1)>' },
        ['<script>alert(1)</script>']
      ],
      info: { title: '<b>Nested</b>', creationDate: new Date(0) }
    });
    expect(Buffer.from(buffer).toString('base64')).toMatchSnapshot();
  });

  it('creates employee report', async () => {
    const def = buildEmployeeReport({
      employee: { firstName: '<b>Alice</b>', lastName: 'Smith', id: '1' },
      events: [{ title: '<i>Bonus</i>', eventDate: new Date(0) }]
    });
    def.info = { ...(def.info || {}), creationDate: new Date(0) };
    const buffer = await pdfBuffer(def);
    expect(Buffer.from(buffer).toString('base64')).toMatchSnapshot();
  });

  it('adds profile image when provided', () => {
    const def = buildEmployeeReport({
      employee: {
        firstName: 'Alice',
        lastName: 'Smith',
        id: '1',
        profileImage: 'data:image/png;base64,AAAA',
      },
      events: [],
    });
    expect(
      Array.isArray(def.content) &&
        def.content.some((c: any) => typeof c === 'object' && 'image' in c)
    ).toBe(true);
  });

  it('creates employee history report', async () => {
    const def = buildEmployeeHistoryReport([
      { firstName: '<b>Alice</b>', lastName: 'Smith', id: '1' },
      { firstName: 'Bob', lastName: '<i>Jones</i>', id: '2' },
    ]);
    def.info = { ...(def.info || {}), creationDate: new Date(0) };
    const buffer = await pdfBuffer(def);
    expect(Buffer.from(buffer).toString('base64')).toMatchSnapshot();
  });
});
