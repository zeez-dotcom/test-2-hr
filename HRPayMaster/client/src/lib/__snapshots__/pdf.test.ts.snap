// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`pdf utility > generates pdf buffer 1`] = `"JVBERi0xLjMKJf////8KOCAwIG9iago8PAovVHlwZSAvRXh0R1N0YXRlCi9jYSAxCi9DQSAxCj4+CmVuZG9iago3IDAgb2JqCjw8Ci9UeXBlIC9QYWdlCi9QYXJlbnQgMSAwIFIKL01lZGlhQm94IFswIDAgNTk1LjI4IDg0MS44OV0KL0NvbnRlbnRzIDUgMCBSCi9SZXNvdXJjZXMgNiAwIFIKPj4KZW5kb2JqCjYgMCBvYmoKPDwKL1Byb2NTZXQgWy9QREYgL1RleHQgL0ltYWdlQiAvSW1hZ2VDIC9JbWFnZUldCi9FeHRHU3RhdGUgPDwKL0dzMSA4IDAgUgo+PgovRm9udCA8PAovRjEgOSAwIFIKPj4KL0NvbG9yU3BhY2UgPDwKPj4KPj4KZW5kb2JqCjUgMCBvYmoKPDwKL0xlbmd0aCAxNTMKL0ZpbHRlciAvRmxhdGVEZWNvZGUKPj4Kc3RyZWFtCnicrY+9CgJBDIT7PEVewL3M/e0uiMWhHtgp24mFLHdioSALPr9RtlA8rCR8KYaZMAGLzgy6XA3jPMcLFX0CnxIVy+F+jsOu7zgmkpc1xSuNdCNMBbuQdXAtbL0Y21g4x0GPrsEoOYy0n4sIlFKplFppsta+aVZx2ffET+SOC5YDhw2tAm3/0hwVTOt9Cf/7g6/GH00eqPlIvAplbmRzdHJlYW0KZW5kb2JqCjExIDAgb2JqCihwZGZtYWtlKQplbmRvYmoKMTIgMCBvYmoKKHBkZm1ha2UpCmVuZG9iagoxMyAwIG9iagooRDoxOTcwMDEwMTAwMDAwMFopCmVuZG9iagoxNCAwIG9iagooVGVzdCkKZW5kb2JqCjEwIDAgb2JqCjw8Ci9Qcm9kdWNlciAxMSAwIFIKL0NyZWF0b3IgMTIgMCBSCi9DcmVhdGlvbkRhdGUgMTMgMCBSCi9UaXRsZSAxNCAwIFIKPj4KZW5kb2JqCjE2IDAgb2JqCjw8Ci9UeXBlIC9Gb250RGVzY3JpcHRvcgovRm9udE5hbWUgL0JaWlpaWitSb2JvdG8tUmVndWxhcgovRmxhZ3MgNAovRm9udEJCb3ggWy03MzcuMzA0Njg3IC0yNzAuOTk2MDk0IDExNDguOTI1NzgxIDEwNTYuMTUyMzQ0XQovSXRhbGljQW5nbGUgMAovQXNjZW50IDkyNy43MzQzNzUKL0Rlc2NlbnQgLTI0NC4xNDA2MjUKL0NhcEhlaWdodCA3MTAuOTM3NQovWEhlaWdodCA1MjguMzIwMzEzCi9TdGVtViAwCi9Gb250RmlsZTIgMTUgMCBSCj4+CmVuZG9iagoxNyAwIG9iago8PAovVHlwZSAvRm9udAovU3VidHlwZSAvQ0lERm9udFR5cGUyCi9CYXNlRm9udCAvQlpaWlpaK1JvYm90by1SZWd1bGFyCi9DSURTeXN0ZW1JbmZvIDw8Ci9SZWdpc3RyeSAoQWRvYmUpCi9PcmRlcmluZyAoSWRlbnRpdHkpCi9TdXBwbGVtZW50IDAKPj4KL0ZvbnREZXNjcmlwdG9yIDE2IDAgUgovVyBbMCBbOTA4IDYyMi4wNzAzMTMgMjQzLjE2NDA2MyAzMjcuMTQ4NDM4IDIxMS40MjU3ODEgNTYxLjUyMzQzOCA1NjEuNTIzNDM4IDU1MC43ODEyNSA1MzAuMjczNDM4IDU3MC4zMTI1IDQxMi41OTc2NTZdXQovQ0lEVG9HSURNYXAgL0lkZW50aXR5Cj4+CmVuZG9iagoxOCAwIG9iago8PAovTGVuZ3RoIDI1MgovRmlsdGVyIC9GbGF0ZURlY29kZQo+PgpzdHJlYW0KeJxdkDtvwyAQx3c+xY3pEDlx2qSDhRQli4c+VLdT1QHDYSHVgDAe/O0LR5pURYKf7vHnHtWpPbfWRKheg5MdRtDGqoCTm4NE6HEwlm1rUEbGi0WvHIVnVRJ3yxRxbK120DQMoHpL4SmGBVZH5Xq8y76XoDAYO8Dq49SRp5u9/8YRbYQN4xwU6vTdk/DPYkSoSLpuVYqbuKyT6pbxvniEmuxtaUk6hZMXEoOwA7Jmkw5vdDqcoVX/whdRr/9mQ4bg8Hkz6z1hLwmHe8KuL8664FDwWPBQoItc869c/bdObiQv7TqknENI89FmabA8krF4Xb53Pqvo/gATqIIGCmVuZHN0cmVhbQplbmRvYmoKOSAwIG9iago8PAovVHlwZSAvRm9udAovU3VidHlwZSAvVHlwZTAKL0Jhc2VGb250IC9CWlpaWlorUm9ib3RvLVJlZ3VsYXIKL0VuY29kaW5nIC9JZGVudGl0eS1ICi9EZXNjZW5kYW50Rm9udHMgWzE3IDAgUl0KL1RvVW5pY29kZSAxOCAwIFIKPj4KZW5kb2JqCjQgMCBvYmoKPDwKPj4KZW5kb2JqCjMgMCBvYmoKPDwKL1R5cGUgL0NhdGFsb2cKL1BhZ2VzIDEgMCBSCi9OYW1lcyAyIDAgUgo+PgplbmRvYmoKMSAwIG9iago8PAovVHlwZSAvUGFnZXMKL0NvdW50IDEKL0tpZHMgWzcgMCBSXQo+PgplbmRvYmoKMiAwIG9iago8PAovRGVzdHMgPDwKICAvTmFtZXMgWwpdCj4+Cj4+CmVuZG9iagoxNSAwIG9iago8PAovTGVuZ3RoIDQzNTMKL0ZpbHRlciAvRmxhdGVEZWNvZGUKPj4Kc3RyZWFtCnicjTcLcBRHdj2zsz/tSuxKWglYQc8y7Aoz+ku7EhJIg6RdIQmhz0p4RxKwq9UPEAgWIckfjMBn4Mb8fBhjbGPHLscVH5VkFp/thbrz2VX23Tku5yp2Ukkld85VpSpVd3GVc3e5S4UzTF73jITgnKvsds/0+/R7r1+/97pnJnV0DNnRAjIh1+RYYhTpv5eghyYBYcCfQV8/NZ1chH8PvehAYv6QDjK58OCTszO8AVfB4+6h1JhBZyug/+3E1CPjOszdRcisTB6YmdfhrDPwqBk/NHHAgL+P0IpTiIGhif27x15+7eqeFZt/h7w2Sv3wP91l5P1v/+0Zvr36zj/YK21HALQjVleOkHXqrhMQHbdX3+2wV1I5y3/Z0KvQG+gW+hX6PdPOpJlfsiH2cfYie539iHJnozeRB80iK8h0oXL0Atj/7yuOgpcYC0Ies4rySefqUT5C2peL/W5c+x3gPTD+D5DyA/Q+ugb/W+gK/IspdAp9H10HzAvoc/QWusr8Aj1HoTcZBzqJ/gr9GTqLLqOboL8QXQL8Y+hp9Cr6guKvoXcMKa+DnNdAzjvohzD7JDoPnK+hGyDzGXQV/SVIfo/5mLlDZbeh52GGLv8qegUkq+gEegrkPgvjt4GjEPWgJJpHx9CTgL0I/K+BJX+PvmACoP/bYMc19Ofoe9w55JK0J2qODTzeqOHHGjT86BYNP9Ig4fnNGp7bvAXPBq34aL2GZ+on8JFN3Ti1ScOHQxo+VPcpnq7T8MHa3fhA7ad4qlbD+2tL8b7alXhvjYYnqzQ8UfUpHq/W8FilhkcrNJysuI5HKiZwokzD8bIFvKdcw7vLL+FdpRoeLtHwkKjhwY2fYnmjhmMbs/HDD2l45wYNDxSvx/3FpTjq3437/Bru9V/HPQENdwcW8I71Gu4SRLxduI47BQ13CDtxO8Db1mm4jddwhE/hsE/Drb4UbsEabl57CW9dq2FpjYabijS8peGS9Cu8uaEG1wdz8KZQCteFunFtaB0OBS/h6ioJV1akcHlZBxY3Srh4FYcD/g3YX+1dNbx+dS4WzKtXDa9bpWEfvwXzdSsLhvHKUry2UMNrCjRcVGDF3upVRUMrawqKhlaTUSEZeVY1FrwxmFeZO+CudA3kyi45u8Y5YK7hBpwyJ2NuD8eu4I5zX3GmFbtzBhw1WQPWGssAU4kGcuQs2SIftzDllm7LtMXUZNljOW4xIbkcMeVoGn2FTPYa24Cphh2wyayM2T0su4I9zn7FmkySZGYyzEW1X+zMWLW+TtXeM6QyZ1R/lDyl3kHVckZFA4NDsTTDnJefOncOrWnuVC9GYzdMCIZymmVbemNpznRePjKDRCSKInkfmTlKAALqCDLSn8yyP0EwpBESQ5l15B8PF0d/RNIh+rufZ/G3ElkQVDzTBshZE+S6A61AeZANaKvHF6x2C24f9DwYM4zAuJlqxmS/M+sNsR+GCu/Mssqdv65ne+nDrIZCt3vMKu2nQiHz8O2z5hm9Qz0Z177kfmv+EDWj9VK+kItWrGMsqysqRLe7IS9lXW1KIbFKrCqvqip3F26qqNxaGwoFawLCOovFk19QUAjKy9hgTShUXVVQ4Mm3WKzmHJZQqqsa2WA1QdAmrAsECBvzwj83hYf763s2lOyoYtNf3Pmkc7Zl6PWD+1/sr5Ib7v7rtua8xuPx/lee6Hwj0L3FnmXv/JeDz3ye6n7atOFF8eHtDYmQHCyNT9Z//SP29dglpSN1rW/g6WhkerjkH4PxjoK22bauCz9+JKleabSvtPf27rs1P3z2UVJ34dRgdkJ9NCHn2yaUYrNgYSKsB7zI7LxlQbfB31BIkV37kn3YvBX4PChPsme5HSmONVmIG8rFKpgALs+rdtMVhmoLjZWxfXfe/dHNdSFcWJ3NFLRVtaZauPrz5xnrHz6prcu1m65Ys0PHp5CmoYfu/pzp58JcMXJoX5iOgFank1h1FvbBByZwaDXU9FVSjq/AmhIdoB0sRcRYsg8VldFqt1VgPPk5rLCujC0ONpqqq9ay1mBNGSByWKtnLUtcf/aW67umy3VTL41NvLivyels2vfC+PjL++turW2e6o5MbCt1uNofG+rc38LD0j/7jA14Zj9+tq/t1Htzw/PvnW7r/s4nz3h6LxxoquidqP115ETflqnzxMrE3QHTb7gdEIkPoUbklVZU+B2p+tXWVI5lRQoRLxErqZ/AR7pzimupr0JBNxOsaWSJuYWeHBMxv9aTT8wNhWqDZSZivqnjL2Z+erruyjOVidbL7bGFtjsbGqZfTo5f3d/odDTtf2Eice1Qw01feKpzeL4t31HaNhnZfiDs4+r/8PMLf7NXiEYO9knNJ3abGu9+5Hn8k+90t51+b3547r1TbX3Pfvy0p/f81JYdC52/rp3orWg6cAF2+xws6hc0KlajQsnpZguyrKml8BCr3NV6iOQV6372wLgmECCmMpGb57oYaebVkZ6FoYpbVfFI+HBvqR5IX0dM0z842yXuvrz/TgP744GJ8vrxJ0kU7tK+NP0W/CciLLkm8hhLXm4uWuWwptYjh55p7k3l1Xt2VVSGwT3BRX95IOh8QX2HPYbPagtzWHbH4XefjOw49d2hmTenar7+J3aaqz64p2nf9o1idG77yJnqQ8LUtfyus++nNn7rpxc7ei58kIpeydl2JhacfG5kY9fp8fqeFjkY3wyeGIb4C4JlHrDN9W4RhH2xxaqbBEF3zxAj8IydayQ26nYNty68czj17onW1hNvp1LvnGi9Geia6dp+ZHtxcdcReHcFWHfh0Y8vR6OXPz5aePQnZPCTs4W9Fw42NR280FvYc2FakqYvEC/laU+a3LTu2W8glkHlTbALUGlM7juvvsLGzer/fG4pBb6L2n8xs3D/c6Cst0x26sCKyrkaUougFEHsXWyoq2sgfaR869byMkki0tdABgbRz1AWsqU5hMo/ragcA3ZrMVkMExyO7uVLCn92eeiT1vaSoUToeWQ6i0a522ic+Q16iX0Z2Zk0eoh7Ap2FnuAm0TmuEe3iLqJhUxfKY4vQReYSWiN1XH3+ynOnTz31rSdPnlg4/sSxxx979JH5udmjM0dShw9NHzwwtX/f3smJ8bHR5Egivmf3ruGhQTn28M6B/mhvT/eOru2dHe3b2iIbsCvLXsKkHVktQstYVmkJSmc5YOgoLWFUS4tqpUi1W+Th2Iv5Ovti4Vavzyd7BZ8qqZw/THpiVEkuEmQQAbNgLojojAqdvYMxPqzEKREw/fdBOr1uiWaMVLalP6ZGRICWwW0UXgK3PUBuXyQLvIp6FGU0jUx+wEveNEMH5panZViJLKgjouATYmPAm7Yhp68/3gIj5+KI4dtAIp9xoRHoyYeFDGOMBmMqHx+XtwE3Yv0qbdEMCgrz+jiu8kmeVy1+YaQnpvhUJi54DbgvBh5jEl7FJ/h4Wc5oHxQRbsEHsljUnBaYM71piTkTHYyBKpU/0x+7wTJsSxwuE+uBFsvwSJUoliVYgiQATwDUycDO3GBtlN+bkZC6QKkcRVA4CaugOJ3ppgSRmsywOs6lKwoQRUBhgcLpFGmRmwOcTcct6NwbDG4bUFyEcpPkkkqJ+g+8BDsjZZklm2SXnGw2C3tBUDcAcwu+NewMesvJZDPeNMjso+gMs5C2S94MldRncC4AJ8EtLOHAcsK2TBDo0xc+cG8FA4Oxt5wI5NMncDSTX2lJOM3uEIV7Yd0bg90Lp5kdYhxCm4Amf5iHsFalaIzwxr0Q8xDdraUlJLr4mDDmFeR0fr5yKAxihHTCEoiLih5kJLQEVz0EpsnfnhQiccIBiQKtHVDJnXxcHYmLMORdESVC4iBBuFFBmjX50wznZxpRI3jK4lSzhLFm1SE0L1GaUJNOsRCKVWhWmQLdz2EhzK/cqySFEYg5qSc24R2XEyBblYSEygnNXqhGzZAhKxlYRDiNdoiwmk6Ium6xZwjSkiyfV5RWPi1xgUQyQeBWH2S6YpCE1lZ52Ywwr6hSIhkHjrBMmSH3ABkWEvwo+BWWC76KCuTCPEjm9A/GFOeoMCqATyVJScCyvXxS9ipykvoY5oNpqLTEfK8eGeWIJVnuT47DAxJhJC6M6AiSjw/iJh5EjAPXcpzQQdTRN0PfSocQHgUO0hOjqglizMePynqQoB5aKf5PJmYZEw97SoUrroZFiDEgAKAp6sT94OQSGCE9Dl4r02NF5QIk1mI+dZ9XnZLFJZaEujDCK7xLqBfIg05uIz2ummGwkEyQcmQhsQeIDkDwsRGIXhAYiSuLEQfTuMCSJvWgeJ9IKKJMP6hm/WQ56kIPH5f5eBywkC8+L6+a4c2PJ0hwkULbo6+nB6o9vBJKFOYikjJe1Qo1fzwxJvigPqskTXXvExs5sA5FYyryKoqgqAyY6I8AM4gPqJZAO3lBOyQKiTHYRKKPT4zRuREwl3qHSPOGBZ8MLKyf+hIcB/VhhDySCkSjuguyzex3K7kKv0mBOrULSiwXSO6Mw0HAu/gIT7c6AZFMnNBOIBkE6Yx2P2GE+bQF1ANiepfVfw9D27SoM9uoVLCsL6b2LLJYaYPBYVFlC+uASBbP9MF5wtGNIs4z+9vBvRJElZfM5lW2P2ZsD53fTqZ6FzdMnwYYWmjJQehbtNeh26srtdDmpM3uV21+2GiVAxt0spUs514QwBiM1ueYqLn6AmAMqniDQhcSNwDOP0bXpB+APCmYcDVICKR7M9r7PXACxwXSZZmot1FFZAYVreiCibsshPhNrjA06c1BWjtdwnJ0Fm1WajOh6Usy3+94w3tgleE5n/EjMUNWedrISiPvxrzqpCyO6rMsRgXnoaJC5U720vvFEGSD4LNCHYPlQ1bxalSEY4Ou7bTu1Q69OpCoZCICikAMGQNUgFQkbGPIA0FqCdtUFsClkXCDRYxNqCMvu1CXZhkrVHtSjFzZTij0SjI+qh/N4GVU591MLkMWutF2urezpDT1x8xeTqYhE1DnRCOK9eesuESfIzlpXfSkjdCUJaKZipvTYyNgPGdF2zfOUmz/P2U2YzdVO6WRahSw/WlVJn2DOvTt6mB1yR16neggOa0opLSld+WQDHUG3IDPBdM2gZGbDCvBN4+DKT1EtY1iKAjpZiXm6NvmdwDBBbwf6KHtAKILrPnAq3NBy2gatVvn1p0Admf59Tg3yMZsPTrnRBlGEdLjwBIh3cgkh5GlzgeqviFe31P7/URhSRg56IUliQRKM0649XJeM2gM8C5wVz31ZwBMBVipTzPWgMFgJgysv15RHIv1n5T/mwhJiF4nkaw8iFCPwX7AXmd/M8X2IDaboo1dzl56E6SRDlktqqOF3F/I2WQnAVAG+3vsI6Pm0OvEMsdQFEnF5diVxPfWxZIwLS7OXfTbOE1pY+4D2P7YMcAST31EThKVgbc54CPdS1xHtZEYnxaNq+0xsrsnqbiTIs/vhXtWCwO3LTgo95KjiifctgAtcgpcePYmErQO0Q+XlXCX6iP3YbjzCy6e2Yw2658/gvFlAWcA549t9m6S4Usio/2ySNZLFQuHPPR+heddbiApfC58WqhPUfcaNIHi4BS3BAwusoKnIDl1PmK9k1U6o+AE8g2WVefNIt91i59Uz4t/isyT+VClMmhSmPcRX2TQbuERuC60CCrPD0NRhKM+g7qKZEWBI1URyPfTzpj+JEQmgyqKyP2A3GWW+NcUwdfZcoSziAReIqNdLyKfSvf0nljSOwd6yUhZVJxBE9+oloQcM6QHHjS6lgyC01jQDeEChm5lWBmED0QgriXqDXsInFMkUylg0CVi0P8Cx/8nCgplbmRzdHJlYW0KZW5kb2JqCnhyZWYKMCAxOQowMDAwMDAwMDAwIDY1NTM1IGYgCjAwMDAwMDE4OTAgMDAwMDAgbiAKMDAwMDAwMTk0NyAwMDAwMCBuIAowMDAwMDAxODI4IDAwMDAwIG4gCjAwMDAwMDE4MDcgMDAwMDAgbiAKMDAwMDAwMDMxMCAwMDAwMCBuIAowMDAwMDAwMTc1IDAwMDAwIG4gCjAwMDAwMDAwNjUgMDAwMDAgbiAKMDAwMDAwMDAxNSAwMDAwMCBuIAowMDAwMDAxNjU5IDAwMDAwIG4gCjAwMDAwMDA2NDYgMDAwMDAgbiAKMDAwMDAwMDUzNSAwMDAwMCBuIAowMDAwMDAwNTYxIDAwMDAwIG4gCjAwMDAwMDA1ODcgMDAwMDAgbiAKMDAwMDAwMDYyMyAwMDAwMCBuIAowMDAwMDAxOTk0IDAwMDAwIG4gCjAwMDAwMDA3MzYgMDAwMDAgbiAKMDAwMDAwMTAwNCAwMDAwMCBuIAowMDAwMDAxMzM0IDAwMDAwIG4gCnRyYWlsZXIKPDwKL1NpemUgMTkKL1Jvb3QgMyAwIFIKL0luZm8gMTAgMCBSCi9JRCBbPGRmZTgwM2NhOTNkZTE1ZmVjNTg5M2QzY2Y2MTE3OTg0PiA8ZGZlODAzY2E5M2RlMTVmZWM1ODkzZDNjZjYxMTc5ODQ+XQo+PgpzdGFydHhyZWYKNjQyMQolJUVPRgo="`;
